packages/auth/src/authClient.ts

```diff
@@ -0,0 +1,92 @@
+import type { default as Netlify } from 'gotrue-js'
+import type { User as NetlifyUser } from 'gotrue-js'
+import type { Auth0Client as Auth0 } from '@auth0/auth0-spa-js'
+
+// TODO: Can also return an Auth0 user which doesn't have a definition.
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface Auth0User {}
+export type { NetlifyUser }
+
+export interface AuthClient {
+  restoreAuthState?(): void | Promise<any>
+  login(options?: any): Promise<any>
```

Comment:
Add TODO to add `options` typing

Reply:
I actually had some trouble here and would love to hear how you would solve this:

Auth0 does not take arguments for the login function, but Netlify does. I thought I could solve this with function overloading, but I realise that this is probably not for functions associated to an object?

So I added this random "options" argument that could be anything, and added a better type definition in `AuthClientNetlify`

Reply:
I see what you're saying. Even if you mark `options` as optional with a type, that still implies the Auth0 version would accept the `options` arg. 

You could do something like this:
```typescript
export interface AuthClient {
  restoreAuthState?(): void | Promise<any>
  login(): Promise<any>
  logout(): void | Promise<void>
}

export type AuthClientAuth0 = AuthClient

export interface AuthClientNetlify extends Omit<AuthClient, 'login'> {
  login(options: {
    email: string
    password: string
    remember?: boolean
  }): Promise<NetlifyUser>
  client: Netlify
}
```
...only problem there is that Netlify Auth Clients will need to do something like 
`(client as AuthClientNetlify).login(args)`

but it's possible a consumer would've already cast it anyway